src/s390x/dasd.rs

```diff
@@ -0,0 +1,336 @@
+/// * `partitions` - configuration strings
+fn make_partitions(dasd: &str, partitions: &[String]) -> Result<()> {
+    let dasd_conf = NamedTempFile::new().chain_err(|| "creating dasd partitions config")?;
+
+    write(dasd_conf.path(), partitions.join("\n"))
+        .chain_err(|| "writing dasd partitions config")?;
+
+    if try_format(dasd, &dasd_conf.path()).is_err() {
+        default_format(dasd)?;
+        try_format(dasd, &dasd_conf.path())?;
+    }
+    Ok(())
+}
+
+/// Sometimes it may happen, that config-based format fails, so we have to perform
+/// an auto-format on the whole disk
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn default_format(dasd: &str) -> Result<()> {
+    eprintln!("Auto-formatting {}", dasd);
+    let status = Command::new("fdasd")
+        .env("LC_ALL", "C")
+        .arg("-a")
+        .arg("-r")
+        .arg(dasd)
+        .stdout(std::process::Stdio::inherit())
+        .status()
+        .chain_err(|| "failed to execute fdasd")?;
+    check(status).chain_err(|| "formatting using entire disk")
+}
+
+/// Format disk using a config file
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+/// * `file` - configuration file
+fn try_format<P: AsRef<Path>>(dasd: &str, file: P) -> Result<()> {
+    eprintln!("Formatting {} ", dasd);
+    let status = Command::new("fdasd")
+        .env("LC_ALL", "C")
+        .arg("--config")
+        .arg(file.as_ref().to_str().unwrap())
+        .arg(dasd)
+        .stdout(std::process::Stdio::inherit())
+        .status()
+        .chain_err(|| "failed to execute fdasd")?;
+    check(status).chain_err(|| format!("formatting based on {}", file.as_ref().display()))
+}
+
+/// Copy each partition from CoreOS raw image to corresponding /dev/<disk-name>
+///
+/// # Arguments
+/// * `dds` - list of dd options.
+fn copy_partitions(dds: &[Opts]) -> Result<()> {
+    for opts in dds {
+        match dd(&opts) {
+            Ok(s) => {
+                eprintln!("{}", s);
+            }
+            Err(e) => {
+                return Err(e.to_string().into());
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Return disk config
+///
+/// # Arguments
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+fn get_disk_config(dasd: &str) -> Result<DiskConfig> {
+    let cmd = Command::new("fdasd")
+        .env("LC_ALL", "C")
+        .arg("-p")
+        .arg(dasd)
+        .output()
+        .chain_err(|| format!("executing fdasd -p {}", dasd))?;
+    let output = cmd.stdout.iter().map(|&c| c as char).collect::<String>();
+    let config = DiskConfig::from_str(&output)?;
+    eprintln!("{}", config);
+    Ok(config)
+}
+
+impl FromStr for DiskConfig {
+    type Err = Error;
+
+    fn from_str(data: &str) -> Result<Self> {
+        fn get_value(s: &str, data: &str) -> Result<u32> {
+            let pt = format!(r"{}\D+(?P<v>\d+)", s);
+            Regex::new(&pt)
+                .chain_err(|| format!("compiling {}", pt))?
```

Comment:
Personally for a regexp that's static in the code I think it's ok to use `.unwrap()` or `.expect("regex")`.  (But this is fine as is!)